spacy/tokens/underscore.py

```diff
@@ -88,8 +95,46 @@ def get(self, name: str) -> Any:
     def has(self, name: str) -> bool:
         return name in self._extensions
 
-    def _get_key(self, name: str) -> Tuple[str, str, Optional[int], Optional[int]]:
-        return ("._.", name, self._start, self._end)
+    def _get_key(
+        self, name: str
+    ) -> Union[
+        Tuple[str, str, Optional[int], Optional[int]],
+        Tuple[
+            str,
+            str,
+            Optional[int],
+            Optional[int],
+            Optional[Union[str, int]],
+            Optional[Union[str, int]],
+            Optional[Union[str, int]],
+        ],
+    ]:
```

Comment:
As the types have changed to just `int`, this can be (luckily) shortened quite a bit:
```suggestion
    ) -> Union[
        Tuple[str, str, Optional[int], Optional[int]],
        Tuple[str, str, Optional[int], Optional[int], int, int, int],
    ]:
```